### ISSUE TITLE
Data Processing

### ISSUE THREAD

**Issue by cicdguy** [insightsengineering/tern#47]
Basic/generic data pre-processing should not be done on the user but rather on the back end. this is true for both tern and teal but opening issue in tern to cover both.

1.  changing study variable type to conform to tern/teal variable type requirements on the front end is going to diminish adoption. if tern/teal needs a factor variable then conversion should happen in the tern package. many examples in agile-R do not work when pointed at study data.
2. having to remove records with missing values in key variables is a requirement that should not be done on the front end by the user. this should data management should be done in the tern package. example is the AE table where user has to remove records where  coded variables and others like grade have missing values. this should be done in the tern package if that is what is needed. would add that this will under report AE because there is no section in the table displaying how many AEs are not coded which happens when a trial is ongoing. 

Provenance: 
```
Creator: npaszty
```

**Comment by shajoezhu** [comment]
i think we can close this issue now. as preprocessing is no longer tern's job. we do provide some helper functions though. @pawelru @khatril close?

not sure for teal, if it is more appropriate, we can move this issue to teal, @donyunardi , what do you reckon?

**Comment by donyunardi** [comment]
I don't believe the main teal framework (teal, teal.slice, teal.data, teal.reporter, teal.transform) performs any data processing that being mentioned in the original post. 

However, point 1 made me recall this issue where a specific variable type is required to conform to a teal module:
https://github.com/insightsengineering/teal.modules.clinical/issues/687

I do agree that the conversion of data type to conform with the module should be done in the backend. 
I think we should evaluate the teal modules functions using a dataset with character type for all columns and see how it behaves.
